src/plugins/configuration_provider/server/provider.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IScopedClusterClient, Logger } from '../../../../src/core/server';
+
+import { ConfigurationClient } from './types';
+
+const CSP_RULES_DOC_ID = 'csp.rules';
+
+export class OpenSearchConfigurationClient implements ConfigurationClient {
+  private client: IScopedClusterClient;
+  private configurationIndexName: string;
+  private readonly logger: Logger;
+
+  constructor(
```

Comment:
I think we can make this a service and then plugin developers can make this an optional plugin within their repo and then call to the DynamicConfigService(key, value) to insert or DynamicConfigService(key) to get value. Then we can get other plugins onboarded if they want to with little effort.

Reply:
I am evaluating this.